tests/TestCase/Validation/ValidationTest.php

```diff
@@ -2813,6 +2817,9 @@ public function testContainNonAlphaNumeric()
         $this->assertFalse(Validation::containsNonAlphaNumeric('abcdef##', 3));
         $this->assertFalse(Validation::containsNonAlphaNumeric('abc##def', 3));
         $this->assertFalse(Validation::containsNonAlphaNumeric('ab#cd#ef', 3));
+
+        //CHECK SCALAR
```

Comment:
I think so. Cause someone might be tried with Scalar value (CC or AlphaNumeric).